### PR TITLE
Fix price error field name in mutation responses

### PR DESCRIPTION
**Pull request by tomaszszymanski129**
I want to merge this change because it fixes backend responding with `priceAmount` intead of `price` on mutations:
- `ShippingPriceCreate`
- `ShippingPriceUpdate`
- `ProductVariantCreate`
- `ProductVariantUpdate`

Input of both mutations accepts price and error of the same name should be returned. Checked on frontend to make sure it now marks the field red as it should.

Linked task: [SALEOR-1337](https://app.clickup.com/t/2549495/SALEOR-1337)

# Impact

* [ ] New migrations
* [ ] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [ ] Documentation needs to be updated

# Pull Request Checklist

* [ ] Privileged queries and mutations are guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [ ] Database migration files are up to date
* [X] The changes are tested
* [X] GraphQL schema and type definitions are up to date
* [ ] Changes are mentioned in the changelog

### CODE REVIEW

File: saleor/graphql/product/tests/test_variant.py [hunk 1]
```diff
@@ -666,6 +666,51 @@ def test_update_product_variant_unset_cost_price(
...
+                        costPrice: $costPrice,
+                    }) {
+                    productErrors {
+                        field
+                        message
+                        code
+                    }
+                }
+            }
+
```
Comment (fowczarek) [inline saleor/graphql/product/tests/test_variant.py:692]:
```suggestion
```

File: saleor/graphql/shipping/mutations.py [hunk 2]
```diff
@@ -164,6 +164,15 @@ class Meta:
 
 
 class ShippingPriceMixin:
+    @classmethod
+    def clean_instance(cls, info, instance):
+        try:
+            cleaned_instance = super().clean_instance(info, instance)
+        except ValidationError as validation_error:
+            cls.remap_error_fields(validation_error, {"price_amount": "price"})
```
Comment (fowczarek) [inline saleor/graphql/shipping/mutations.py:172]: We should add tests for these changes.

**Review by fowczarek — COMMENTED** [review]
I'm not sure that is a good solution. In the feature, if I want to use `price_amount` as an error field it would be renamed to `price`.

**Review by maarcingebala — APPROVED** [review]

**Review by IKarbowiak — APPROVED** [review]

**Review by IKarbowiak — APPROVED** [review]

**Review by maarcingebala — APPROVED** [review]

**Comment by patrys** [comment]
Task linked: [SALEOR-1337 Mutations should always return in "price" field instead of "priceAmount"](https://app.clickup.com/t/2549495/SALEOR-1337)

**Review by fowczarek — APPROVED** [review]

**Pull request merged**